client-ts/tsconfig-base.json

```diff
@@ -13,6 +13,6 @@
         "noImplicitAny": true,
         "suppressImplicitAnyIndexErrors": true,
         "noEmitOnError": true,
-        "lib": [ "es2016", "dom" ]
+        "lib": [ "es5", "es2015.promise", "es2015.iterable", "dom" ]
```

Comment:
We want to use the `es5` lib, but:
* We **do** Polyfill `Promise`, so we want to keep that
* `es2015.iterable` enables `for (const item of list)` syntax, and TypeScript **will** transpile this down to a standard `for(let i = 0; i < list.length; i+=1)` loop, so we still want to use it.

Reply:
What's the side effect of this?

Reply:
1. We can only use types available on ES5 browsers.
2. Exception to the above: We can use `Promise` (and we include a polyfill for that already)
3. Exception to the above: The ECMAScript 2015 [Iteration protocols](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) are available, meaning we can use `for (const item of array)` syntax (`foreach`-like) on the built-in iterable types (`Array`, `[Type]Array`, and also `Set` and `Map` if we could access those types)

Reply:
The `lib` property basically sets standard TypeScript declarations we want auto-imported into our TypeScript files.

Reply:
Does this make our js library bigger than it was before?